### PR TITLE
docs: change from pdoc to sphinx

### PR DESCRIPTION
**Pull request by tmac1973**
Summary:

The old doc generator (pdoc) was not optimal. Sphinx generates much prettier and complete auto documentation. 

Testing:

Tested locally on command line using new poe the poet commands

Issue:

https://lacework.atlassian.net/browse/GROW-2624

### CODE REVIEW

**Review by ipcrm — COMMENTED** [review]
I see a lot of these doctree files are binaries; do we have to check all that in?

File: docs-source/conf.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import os
+import sys
+sys.path.insert(0, os.path.abspath('../laceworksdk'))
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'Lacework Python SDK'
+copyright = '2023, Jon Stewart, Tim MacDonald'
```
Comment (ipcrm) [inline docs-source/conf.py:14]: Should this be Lacework?

File: pyproject.toml [hunk 2]
```diff
@@ -43,14 +43,18 @@ types-requests = "^2.28"
...
 [tool.poe.tasks]
-test        = "poetry run pytest -m 'not ci_exempt and not flaky_test' tests/api/v2"
-flaky_test  = "poetry run pytest -m 'flaky_test' tests"
-lint        = "poetry run flake8 . --exclude 'jupyter' --count --select=E9,F63,F7,F82 --show-source --statistics"
\ No newline at end of file
+test        = "pytest -m 'not ci_exempt and not flaky_test' tests/api/v2"
+flaky_test  = "pytest -m 'flaky_test' tests"
+lint        = "flake8 . --exclude 'jupyter' --count --select=E9,F63,F7,F82 --show-source --statistics"
+docs-clean.shell  = "rm -r docs-source/auto; rm -r docs/*"
+docs-build.shell  = "sphinx-apidoc -e -f -d 5 -o docs-source/auto laceworksdk; sphinx-build -M html docs-source docs"
```
Comment (ipcrm) [inline pyproject.toml:59]: do we need to handle the errors here? (seeing two commands)

**Review by ipcrm — COMMENTED** [review]
Looks great! Couple quick questions

File: docs-source/conf.py (hunk 1 above)
Comment (tmac1973) [reply docs-source/conf.py:14]: Oh lol yeah prolly so....

File: pyproject.toml (hunk 2 above)
Comment (tmac1973) [reply pyproject.toml]: Good catch

**Review by ipcrm — APPROVED** [review]
Wonderful!

**Pull request merged**